### ISSUE TITLE
Do not add a generic plugin-to-editor extension API

### ISSUE THREAD

**Issue by vbmacher** [emustudio/emuLib#49]
## Resolution

Closed as not planned. This issue combines two unrelated, unspecified workflows and therefore cannot define an emuLib API.

## Current code facts

- `application/src/main/java/net/emustudio/application/gui/editor/Editor.java` opens and saves only host `Path` files through `openFile(Path)`, `saveFile()`, and `saveFileAs()`.
- `plugins/device/88-dcdd/.../CpmFileSystem.java` can `readFile` and `writeFile`, but `writeFileBytes` rejects an existing file with “Overwrites are not supported”.
- `plugins/device/88-dcdd/.../gui/DiskGui.java` has drive selection and mounted-image status only. It has no CP/M file list, CP/M-format selector, or file action.
- The CPU example does not identify which CPU, how the address range is selected, which assembly syntax is produced, or where the command appears.

Adding a generic editor/document/virtual-filesystem API now would require inventing persistence and lifecycle behavior not required by working code.

## Separate ticket required for 88-DCDD

A future 88-DCDD ticket must specify all of:

- supported `CpmFormat` values and how the format is selected;
- file-list UI location in `DiskGui`;
- CP/M filename and user-area handling;
- text encoding and binary-file rejection;
- whether editing is read-only or writable;
- overwrite algorithm, rollback on allocation failure, and timestamp behavior;
- locking while the same mounted drive is used by emulation.

Only after those choices are made should that ticket name the smallest host-editor method it needs.

## Separate ticket required for disassembly

A future CPU ticket must specify all of:

- first CPU plugin;
- start/end address selection;
- output produced from the existing `Disassembler`;
- command location in the debugger;
- read-only versus Save As behavior;
- handling of invalid memory and undecodable instructions.

Do not reopen this umbrella issue for either feature.

**Comment by vbmacher** [comment]
Closing #49 instead of expanding it into a generic UI-extension ticket.

The two examples require different unresolved behavior:

- `application/.../editor/Editor.java` opens and saves host `Path` values only.
- `88-dcdd` has `CpmFileSystem.readFile/writeFile`, but `DiskGui` has no CP/M file browser or CP/M-format selection. `writeFile` explicitly rejects overwrite, so “save back” needs a defined replace/rollback and mounted-drive locking policy before any editor API.
- CPU “decompile to editor” does not identify a CPU, address-range selection, output syntax, UI action, or whether generated text is read-only.

A shared editor port cannot be designed from those two incomplete workflows without inventing a document model, persistence callbacks, and UI lifecycle rules.

Create a new ticket only for one selected workflow:

1. For 88-DCDD: specify supported CP/M formats, file-list UI, text encoding, binary-file handling, overwrite/rollback behavior, and concurrent access while a disk is mounted.
2. For CPU disassembly: specify the first CPU, range-selection UI, exact generated syntax, command location, and save/read-only behavior.

After one workflow is specified, add only the smallest emuLib method required by that implementation.
